src/hotspot/share/utilities/nativeCallStack.cpp

```diff
@@ -96,9 +95,10 @@ void NativeCallStack::print_on(outputStream* out, int indent) const {
         out->print("[" PTR_FORMAT "]", p2i(pc));
       }
 
-      if (Decoder::get_source_info(pc, buf, sizeof(buf), &line_no, frame != 0)) {
-        out->print("  (%s:%d)", buf, line_no);
-      }
+      // Note: we deliberately omit printing source information here. NativeCallStack::print_on()
+      // can be called thousands of times as part of NMT detail reporting, and source printing
+      // can slow down reporting by a factor of 5 or more depending on platform (see JDK-8296931).
```

Comment:
I'm not sure what the convention is, should we still directly refer to bug numbers in comments?